File: apps/web/flush_redis.js
const fs = require('fs');
const path = require('path');
const dotenv = require('dotenv');

// Load environment variables manually
const envPath = path.join(__dirname, '.env.local');
const envConfig = dotenv.parse(fs.readFileSync(envPath));
for (const k in envConfig) {
  process.env[k] = envConfig[k];
}

const url = process.env.UPSTASH_REDIS_REST_URL + '/FLUSHALL';
const token = process.env.UPSTASH_REDIS_REST_TOKEN;

fetch(url, {
  headers: { 'Authorization': `Bearer ${token}` }
}).then(res => res.json())
  .then(data => {
    console.log("Redis flush result:", data);
  })
  .catch(err => {
    console.error("Redis flush error:", err);
  });

File: scripts/flush_redis.py
import os
import requests

env_path = os.path.join(os.path.dirname(__file__), '..', 'apps', 'web', '.env.local')

url = ''
token = ''

with open(env_path, 'r') as f:
    for line in f:
        if line.startswith('UPSTASH_REDIS_REST_URL='):
            url = line.split('=', 1)[1].strip().strip('"').strip("'")
        elif line.startswith('UPSTASH_REDIS_REST_TOKEN='):
            token = line.split('=', 1)[1].strip().strip('"').strip("'")

if url and token:
    res = requests.post(f"{url}/FLUSHALL", headers={"Authorization": f"Bearer {token}"})
    print("Redis flushed:", res.json())
else:
    print("Could not find Redis credentials in .env.local")
